src/ContentValidationListener.php

```diff
@@ -356,6 +367,101 @@ protected function useRawData($controllerService)
         return false;
     }
 
+    /**
+     * @param $controllerService
+     * @return bool
+     */
+    protected function shouldRemoveEmptyData($controllerService)
+    {
+        if (! isset($this->config[$controllerService]['remove_empty_data'])
+            || (isset($this->config[$controllerService]['remove_empty_data'])
+                && $this->config[$controllerService]['remove_empty_data'] === true)
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @param array $data Data to filter null values from
+     * @param array $compareTo Original data, send along to preserve keys/values in $data which are intentional
+     *
+     * @return array
+     */
+    // TODO strict types for PHP > 7 when module gets bumped
+    //    protected function removeEmptyData(array $data, array $compareTo = []) : array
+    protected function removeEmptyData($data, $compareTo = [])
+    {
+        // TODO when bumped to PHP > 7 and strict types above enforced this check may be removed
+        if (! is_array($data) || ! is_array($compareTo)) {
```

Comment:
PHP 5.6 allows us to use the `array` typehint for arguments.; we do not need to wait for PHP 7 here.

Reply:
done